### PR TITLE
Update nightly builds to compile on Ubuntu 24

### PR DESCRIPTION
**Pull request by rsanchez15**
## Description

This PRs:

- updates nightly builds to compile on Ubuntu 24.04,
- removes nightly builds for Ubuntu 22.04, and
- extends the action for installing python packages to setup a virtual environment, 
- adds a new Ubuntu action (`setup_python_venv`) for setting up a python virtual environment.

## Contributor Checklist
- [x] Commit messages follow the company guidelines. 
- [x] Changes are API compatible. 
- [x] New features have been added to the `versions.md` and `README.md` files (if applicable).

## Reviewer Checklist
- [x] The title and description correctly express the PR's purpose.
- [x] The Contributor checklist is correctly filled.

### CODE REVIEW

File: .github/workflows/configurations/metas/ubuntu-24.04/colcon.meta [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (irenebm) [inline .github/workflows/configurations/metas/ubuntu-24.04/colcon.meta:1]: Is it necessary to keep the `ubuntu-20.04/colcon.meta` file?

File: .github/workflows/build_dev_utils.yml [hunk 2]
```diff
@@ -215,6 +215,10 @@ jobs:
         with:
           path: src/eprosima-CI
 
+      - name: Setup Python Virtual Environment
```
Comment (irenebm) [inline .github/workflows/build_dev_utils.yml:218]: Could you consider adding a TODO note for implementing a similar Python virtual environment setup for Windows in the future or explaining that this setup is required for Ubuntu 24.04, and we've included it for Ubuntu 22.04 as well for consistency?

**Review by irenebm — CHANGES_REQUESTED** [review]
Please, update the [README.md](https://github.com/eProsima/eProsima-CI/blob/feature/ubuntu-24/README.md#artifacts-uploaded) with the new artifact names and remove the outdated ones.

File: .github/workflows/configurations/metas/ubuntu-24.04/colcon.meta (hunk 1 above)
Comment (rsanchez15) [reply .github/workflows/configurations/metas/ubuntu-24.04/colcon.meta:1]: Not really. Removed

**Review by irenebm — APPROVED** [review]
LGTM

**Review by irenebm — APPROVED** [review]
LGTM

File: ubuntu/install_python_packages/action.yml [hunk 3]
```diff
@@ -36,16 +52,19 @@ runs:
...
 
         # Install requirements file if any
         if [[ ! -z "${{ inputs.requirements_file_name }}" ]] ; then
-          pip3 install ${UPGRADE_FLAG} --user \
+          pip3 install ${UPGRADE_FLAG} \
             -r ${{ inputs.requirements_file_name }}
         fi
 
+        python -m ensurepip --upgrade
+        python -m pip install --upgrade setuptools
```
Comment (EduPonz) [inline ubuntu/install_python_packages/action.yml:66]:
```suggestion
        python3 -m ensurepip --upgrade
        python3 -m pip3 install --upgrade setuptools
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: ubuntu/install_python_packages/action.yml (hunk 3 above)
Comment (rsanchez15) [reply ubuntu/install_python_packages/action.yml]: Tested in Fast DDS CI

File: versions.md [hunk 4]
```diff
@@ -35,6 +35,10 @@ The [Forthcoming](#forthcoming) section includes those features added in `main`
 
 The upcoming release will include the following **features**:
 
+- Update nightly builds to compile on Ubuntu 24.04.
+- Remove nightly builds for Ubuntu 22.04.
```
Comment (EduPonz) [inline versions.md:39]:
```suggestion
- Remove nightly builds for Ubuntu 20.04.
```

File: versions.md [hunk 5]
```diff
@@ -35,6 +35,10 @@ The [Forthcoming](#forthcoming) section includes those features added in `main`
...
+- Extended the action for installing python packages to setup a virtual environment.
```
Comment (EduPonz) [inline versions.md:40]: Missing the item for the new Ubuntu `setup_python_venv` action

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Pull request merged**